# HG changeset patch
# Parent ae01fec9549d5bc10bf6a6053ca91e129d3a57c3
Call _thr_check_init() from _pthread_once

diff -r ae01fec9549d lib/libthr/thread/thr_once.c
--- a/lib/libthr/thread/thr_once.c
+++ b/lib/libthr/thread/thr_once.c
@@ -64,6 +64,8 @@
 	struct pthread *curthread;
 	int state;
 
+	_thr_check_init();
+
 	for (;;) {
 		state = once_control->state;
 		if (state == ONCE_DONE)
